src/LrtSquare.sol

```diff
@@ -356,6 +379,47 @@ contract LrtSquare is
         return (totalValue * vaultTokenShares) / totalSupply;
     }
 
+    function communityPause() external payable whenNotPaused {
+        if (depositForCommunityPause == 0) revert CommunityPauseDepositNotSet();
+        if (msg.value != depositForCommunityPause)
+            revert IncorrectAmountOfEtherSent();
+
+        _pause();
+        communityPauseDepositedAmt = msg.value;
+        emit CommunityPause(msg.sender);
+    }
+
+    function pause() external onlyRole(PAUSER_ROLE) whenNotPaused {
+        _pause();
+    }
+
+    function unpause() external onlyRole(PAUSER_ROLE) whenPaused {
+        if (communityPauseDepositedAmt != 0)
```

Comment:
can't it unpause when paused by `pause` instead of `communityPause`?

Reply:
The pauser would have to withdraw the community pause deposit amount from the contract before they unpause